### DISCUSSION TITLE
Passing over a trial if a certain amount of time has passed

### DISCUSSION THREAD

**Discussion by dariusmargineansrx** [optuna/optuna#4481] [Q&A] [answered]
Hi!

I have a script that runs over 1000 dfs and computes hyperparameter optimization for all of them (individually) using a Prophet model. 

The issue is that after a certain combination of hyperparameters gets into the model, the trial freezes (I suspect this being a Prophet issue with some values for hyperparameters). However, I don't quite know which are the combinations that make this happen. 

I was wondering if there is an Optuna parameter that can be set in order to pass over that trial if a certain amount of time has passed. 

Here is what I am using atm:

`

    def objective(trial):
        .
        .
        .
        .
        .
        return accuracy

    # Find the best parameters
    optuna_prophet = optuna.create_study(
        direction="minimize", sampler=TPESampler(seed=321)
    )

    optuna_prophet.optimize(objective, n_trials=300)

    prophet_trial = optuna_prophet.best_trial
    prophet_trial_params = prophet_trial.params

    return prophet_trial_params
`

**Comment by nzw0301** [comment, marked as answer]
> I was wondering if there is an Optuna parameter that can be set in order to pass over that trial if a certain amount of time has passed.

I think, no.

**Reply by nzw0301** [reply]
https://github.com/facebook/prophet/issues/1944  might be a related issue on prophet side.

**Reply by dariusmargineansrx** [reply]
Yeah, it is a Prophet issue, but I was wondering if I can handle it using Optuna. Anyway, thanks for the info!
